app/models/reservation.rb

```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+
+class Reservation < ApplicationRecord
+  belongs_to :user
+  belongs_to :reservation_status
+  belongs_to :ticket_desk, optional: true
+  belongs_to :screening
+  has_many :seats, dependent: :destroy
```

Comment:
Are you sure that you want to remove seats from the CinemaHall when the reservation is destroyed?

Reply:
Actually, I made a mistake here: it should be `has_many :seats_reservations, dependent: :destroy `as this refers only to the reserved seats. I will correct it, thank you for spotting this :)